SYCL/ESIMD/api/functional/ctors/ctor_broadcast.hpp

```diff
@@ -0,0 +1,196 @@
+//===-- ctor_broadcast.hpp - Functions for tests on simd broadcast constructor
+//      definition. -------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd broadcast constructor.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "common.hpp"
+
+namespace esimd = sycl::ext::intel::experimental::esimd;
+
+namespace esimd_test::api::functional {
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct initializer {
+  static std::string get_description() { return "initializer"; }
+
+  template <typename SrcT, typename DstT, int NumElems>
+  static void call_simd_ctor(DstT ref_data, SrcT *const out) {
```

Comment:
```suggestion
  static void call_simd_ctor(DstT ref_value, SrcT *const out) {
```
It's more like a single value than a whole reference data generated. What do you think?

Reply:
`ref_value` mean that we have a single value, and in other tests we have `ref_data`